src/Geom_Poly.f90

```diff
@@ -361,20 +373,141 @@ ELEMENTAL FUNCTION isCircle_PolygonType(thisPoly) RESULT(bool)
+      ! REAL(SRK) :: dist,dist_old
+      bool=.FALSE.
+      IF(thisPoly%isinit) THEN
+        IF(thisPoly%isCircle()) THEN
+          bool=.TRUE.
+        ELSE
+          !nVert should be equal to the total number of edges
+          nNonQuadEdges=thisPoly%nVert-thisPoly%nQuadEdge
+          bool=(thisPoly%nQuadEdge == 1) .AND. (nNonQuadEdges == 2)
+          !The following code can be used to check if the lengths of the
+          !non-quadratic edges are equal. This might become relevant if the
+          !section is required to be "ideal". Leaving commented now for speed.
+          ! dist_old=-1.0_SRK
+          ! IF(ALLOCATED(thisPoly%quad2edge)) THEN
+          !   DO i=1,thisPoly%nVert
+          !     IF(.NOT.ANY(thisPoly%quad2edge == i)) THEN
+          !       dist=Distance(thisPoly%vert(thisPoly%edge(1,i)), &
+          !         thisPoly%vert(thisPoly%edge(2,i)))
+          !       IF(dist_old > 0.0_SRK) THEN
+          !         bool=bool .AND. (dist .APPROXEQA. dist_old)
+          !       ELSE
+          !         dist_old=dist
+          !       ENDIF
+          !     ENDIF
+          !   ENDDO
+          ! ENDIF
+        ENDIF
+      ENDIF
+    ENDFUNCTION isSection_PolygonType
+!
+!-------------------------------------------------------------------------------
 !> @brief This routine will query a polygon type, check if it meets the criteria
-!>        to be a circle, and if so it will return the radius for the circle.
+!>        to be a circle or circle section, and if so it will return the radius.
 !> @param thisPoly The polygon type from which to get the radius
-!> @param r The radius if thisPoly is a circle.  It will be 0.0 for all other
-!>        cases.
+!> @param r The radius if thisPoly is a circle or circle section. It will be 0.0
+!>         for all other cases.
 !>
     ELEMENTAL FUNCTION getRadius_PolygonType(thisPoly) RESULT(r)
       CLASS(PolygonType),INTENT(IN) :: thisPoly
       REAL(SRK) :: r
       r=0.0_SRK
-      IF(thisPoly%isCircle()) r=thisPoly%quadEdge(3,1)
+      IF(thisPoly%isSection()) r=thisPoly%quadEdge(3,1)
     ENDFUNCTION getRadius_PolygonType
 !
 !-------------------------------------------------------------------------------
+!> @brief This routine will query a polygon type, check if it meets the criteria
+!>        to be an arc sector, and if so it will return the inner radius.
+!> @param thisPoly The polygon type from which to get the inner radius
+!> @param r The inner radius if thisPoly is an arc sector.  It will be 0.0 for
+!>        all other cases.
+!>
+    ELEMENTAL FUNCTION getInnerRadius_PolygonType(thisPoly) RESULT(r)
+      CLASS(PolygonType),INTENT(IN) :: thisPoly
+      REAL(SRK) :: r
+      INTEGER(SIK) :: i
+      r=0.0_SRK
+      IF(thisPoly%isSection()) THEN
+        r=0.0_SRK
+      ELSEIF(thisPoly%isSector()) THEN
+        r=HUGE(1.0_SRK)
+        DO i=1,thisPoly%nQuadEdge
+          IF(thisPoly%quadEdge(3,i) < r) r=thisPoly%quadEdge(3,i)
```

Comment:
Just want to raise the point that this is a comparison of floats. I think with the `HUGE(1.0_SRK)` you should be ok though.

Reply:
Good point, but I'm changing to use `MAXVAL` anyway.